Description: drop features depending on packages not in Debian yet
 Disable all providers except for mbed-crypto and tpm. Namely:

 - pkcs11-provider
 - trusted-service-provider
 - cryptoauthlib-provider

 In theory, all crates needed by trusted-service-provider are packaged in
 Debian. However, it needs files from another arm project called
 trusted-services-vendor, which has not been packaged yet.

 Also disable the jwt-svid authenticator.

 We will enable the missing features incrementally once the needed crates enter
 Debian.
Author: Emanuele Rocca <ema@debian.org>
Last-Update: 2023-08-09
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
Index: parsec-service/Cargo.toml
===================================================================
--- parsec-service.orig/Cargo.toml
+++ parsec-service/Cargo.toml
@@ -29,10 +29,8 @@ repository = "https://github.com/paralla 1
 
 [package.metadata.docs.rs]
 features = [
-    "pkcs11-provider",
     "tpm-provider",
     "mbed-crypto-provider",
-    "cryptoauthlib-provider",
     "direct-authenticator",
 ]
 
@@ -49,11 +47,6 @@ version = "0.21.0" 2
 [dependencies.bincode]
 version = "1.3.1"
 
-[dependencies.cryptoki]
-version = "0.6.0"
-optional = true
-default-features = false
-
 [dependencies.derivative]
 version = "2.2.0"
 
@@ -78,7 +71,7 @@ version = "0.2.14" 3
 version = "0.29.1"
 
 [dependencies.picky-asn1]
-version = "0.11.0"
+version = "0.7.0"
 optional = true
 
 [dependencies.picky-asn1-der]
@@ -86,7 +79,7 @@ version = "0.4.0" 4
 optional = true
 
 [dependencies.picky-asn1-x509]
-version = "0.9.0"
+version = "0.10.0"
 optional = true
 
 [dependencies.prost]
@@ -114,4 +107,0 @@ optional = true 5
-[dependencies.rust-cryptoauthlib]
-version = "0.4.5"
-optional = true
-
@@ -125,10 +114,6 @@ features = ["derive"]
 [dependencies.signal-hook]
 version = "0.3.4"
 
-[dependencies.spiffe]
-version = "0.2.1"
-optional = true
-
 [dependencies.structopt]
 version = "0.3.26"
 default-features = false
@@ -151,10 +136,6 @@ features = ["zeroize_derive"]
 version = "0.8.3"
 features = ["small_rng"]
 
-[dev-dependencies.rust-cryptoauthlib]
-version = "0.4.4"
-features = ["software-backend"]
-
 [build-dependencies.bindgen]
 version = "0.66.1"
 optional = true
@@ -167,28 +148,14 @@ optional = true
 all-authenticators = [
     "direct-authenticator",
     "unix-peer-credentials-authenticator",
-    "jwt-svid-authenticator",
 ]
 all-providers = [
     "tpm-provider",
-    "pkcs11-provider",
     "mbed-crypto-provider",
-    "trusted-service-provider",
 ]
-cryptoauthlib-provider = ["rust-cryptoauthlib"]
-default = ["unix-peer-credentials-authenticator"]
+default = ["unix-peer-credentials-authenticator", "mbed-crypto-provider", "tpm-provider"]
 direct-authenticator = []
-jwt-svid-authenticator = ["spiffe"]
 mbed-crypto-provider = ["psa-crypto"]
-pkcs11-provider = [
-    "cryptoki",
-    "picky-asn1-der",
-    "picky-asn1",
-    "picky-asn1-x509",
-    "psa-crypto",
-    "rand",
-    "hex",
-]
 tpm-provider = [
     "tss-esapi",
     "picky-asn1-der",
@@ -196,10 +163,4 @@ tpm-provider = [
     "picky-asn1-x509",
     "hex",
 ]
-trusted-service-provider = [
-    "psa-crypto",
-    "bindgen",
-    "prost-build",
-    "prost",
-]
 unix-peer-credentials-authenticator = []
